### PR TITLE
Case 22249: migrate to Qt 5.12.3

### PR DESCRIPTION
**Pull request by AndrewMeadows**
https://highfidelity.fogbugz.com/f/cases/22249/Migrate-to-Qt-5-12-3

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19409/HighFidelity-Beta-PR15553-2f95c8e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19409/HighFidelity-Beta-Interface-PR15553-2f95c8e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19409/HighFidelity-Beta-PR15553-2f95c8e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19409/HighFidelity-Quest-Beta-PR15553-2f95c8e.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19409/HighFidelity-Beta-PR15553-2f95c8e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19409/HighFidelity-Beta-Interface-PR15553-2f95c8e.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19422/HighFidelity-Beta-PR15553-51657ba.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19422/HighFidelity-Beta-Interface-PR15553-51657ba.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19422/HighFidelity-Beta-PR15553-51657ba.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19422/HighFidelity-Quest-Beta-PR15553-51657ba.apk)

**Comment by AndrewMeadows** [comment]
Hrm... Jenkins appears to have intermittent build failure on macos.  Jenkins, rebuild this please.

**Comment by AndrewMeadows** [comment]
Hrm... Jenkins appears to have intermittent build failure on macos. Jenkins, rebuild this please.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19427/HighFidelity-Beta-PR15553-51657ba.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19427/HighFidelity-Beta-Interface-PR15553-51657ba.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19427/HighFidelity-Beta-PR15553-51657ba.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19427/HighFidelity-Beta-Interface-PR15553-51657ba.exe

**Comment by AndrewMeadows** [comment]
Sigh... if the **macos** build doesn't mysteriously fail it is **ubuntu**.  Trying again: Jenkins rebuild this please.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19437/HighFidelity-Beta-PR15553-51657ba.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19437/HighFidelity-Beta-Interface-PR15553-51657ba.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19437/HighFidelity-Beta-PR15553-51657ba.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19437/HighFidelity-Quest-Beta-PR15553-51657ba.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19437/HighFidelity-Beta-PR15553-51657ba.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19437/HighFidelity-Beta-Interface-PR15553-51657ba.exe

**Comment by AndrewMeadows** [comment]
OMG all builds succeeded!  Quick, let's merge it while it lasts.

File: BUILD_OSX.md [hunk 1]
```diff
@@ -17,16 +17,17 @@ For OpenSSL installed via homebrew, set OPENSSL_ROOT_DIR:
...
-Assuming you've installed Qt using the homebrew instructions above, you'll need to set QT_CMAKE_PREFIX_PATH so CMake can find your installations.
-For Qt installed via homebrew, set QT_CMAKE_PREFIX_PATH:
+Set QT_CMAKE_PREFIX_PATH so CMake can find your installations.
+For Qt installed as described above, set QT_CMAKE_PREFIX_PATH (in *.bash_profile*):
 
-    export QT_CMAKE_PREFIX_PATH=/usr/local/Cellar/qt/5.10.1/lib/cmake
+    export QT_CMAKE_PREFIX_PATH=/Users/<your user>/Qt/5.12.3/clang_64/lib/cmake
 
-Note that this uses the version from the homebrew formula at the time of this writing, and the version in the path will likely change.
+**Note** that this uses the version from the homebrew formula at the time of this writing, and the version in the path will likely change.
```
Comment (SamGondelman) [inline BUILD_OSX.md]: This note doesn't make sense anymore.

File: CMakeLists.txt [hunk 2]
```diff
@@ -203,6 +203,12 @@ find_package( Threads )
 add_definitions(-DGLM_FORCE_RADIANS)
 add_definitions(-DGLM_ENABLE_EXPERIMENTAL)
 add_definitions(-DGLM_FORCE_CTOR_INIT)
+
+if (WIN32)
```
Comment (jherico) [inline CMakeLists.txt]: should also include `MSCV_VERSION < 1920`

File: cmake/macros/SetupQt.cmake [hunk 3]
```diff
@@ -6,33 +6,6 @@
 #  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
 #
 
-# Construct a default QT location from a root path, a version and an architecture
```
Comment (jherico) [inline cmake/macros/SetupQt.cmake]: why are we eliminating this logic?

**Comment by AndrewMeadows** [comment]
I'm going to close this one.  We prefer to take the PR-15510 route where we supply Qt via VCPKG because then we won't have to update Jenkins.

**Pull request closed without merging**